lib/spack/spack/cmd/test.py

```diff
@@ -23,45 +26,67 @@
 
 
 def setup_parser(subparser):
-    subparser.add_argument('--keep-stage', action='store_true',
-                           help='Keep testing directory for debugging')
-    subparser.add_argument(
+    sp = subparser.add_subparsers(metavar='SUBCOMMAND', dest='test_command')
+
+    # Run
+    run_parser = sp.add_parser('run', help=test_run.__doc__)
+
+    run_parser.add_argument(
+        '-n', '--name', help="The name of the test. Default is current time"
+    )
+    run_parser.add_argument('--keep-stage', action='store_true',
+                            help='Keep testing directory for debugging')
+    run_parser.add_argument(
         '--log-format',
         default=None,
         choices=spack.report.valid_formats,
         help="format to be used for log files"
     )
-    subparser.add_argument(
+    run_parser.add_argument(
         '--log-file',
         default=None,
         help="filename for the log file. if not passed a default will be used"
     )
-    arguments.add_cdash_args(subparser, False)
-    subparser.add_argument(
+    arguments.add_cdash_args(run_parser, False)
+    run_parser.add_argument(
         '--help-cdash',
         action='store_true',
         help="Show usage instructions for CDash reporting"
     )
 
-    length_group = subparser.add_mutually_exclusive_group()
+    length_group = run_parser.add_mutually_exclusive_group()
     length_group.add_argument(
         '--smoke', action='store_true', dest='smoke_test', default=True,
         help='run smoke tests (default)')
     length_group.add_argument(
         '--capability', action='store_false', dest='smoke_test', default=True,
         help='run full capability tests using pavilion')
 
-    cd_group = subparser.add_mutually_exclusive_group()
+    cd_group = run_parser.add_mutually_exclusive_group()
     arguments.add_common_arguments(cd_group, ['clean', 'dirty'])
 
-    arguments.add_common_arguments(subparser, ['installed_specs'])
+    arguments.add_common_arguments(run_parser, ['installed_specs'])
 
+    # List
+    list_parser = sp.add_parser('list', help=test_list.__doc__)
+    list_parser.add_argument(
+        'filter', nargs=argparse.REMAINDER,
+        help='optional case-insensitive glob patterns to filter results.')
 
-def test(parser, args):
-    # record test time
-    now = datetime.datetime.now()
-    tty.msg("Spack test at %s" % now.strftime('%Y-%m-%d_%H:%M:%S'))
+    # Status
+    status_parser = sp.add_parser('status', help=test_status.__doc__)
+    status_parser.add_argument('name', help="Test for which to provide status")
 
+    # Status
+    results_parser = sp.add_parser('results', help=test_results.__doc__)
```

Comment:
This feature returns

```===> No test openmpi found in test stage```

when I run `spack test results openmpi` after `spack test run openmpi`.  The directory contains the following despite running the tests after `spack clean -t`:

```
openmpi-2iikagisua64x252c7uk3oc5bpeyvs6v/
openmpi-zjqpy6nclufs6db35w5l6xdn3rkyg5id/
results.txt 
test-openmpi-2iikagi-out.txt 
test-openmpi-zjqpy6n-out.txt 
```

where the contents of `results.txt` are:

```
openmpi-2iikagi FAILED
openmpi-zjqpy6n FAILED
```

Reply:
After off-line discussions, I see I did not understand/intuit the `name` option.